### ISSUE TITLE
Carbs logged in the Omnipod PDM are not imported when the bolus is zero

### ISSUE THREAD

**Issue by alex-gonzalez** [tidepool-org/chrome-uploader#759]
The Omnipod PDM has at least 2 situations where use of the wizard can result in a zero bolus but contains data with direct clinical relevance:
- Carbs are entered and based on BG wizard suggests 0.00 bolus
- Carbs are entered and suggested bolus is reduced by user to 0.00

Because of the filtering in insuletSimulator.getEvents(), although these carb events are logged in the PDM they are never imported into Tidepool.

Some issues this raises in the Tidepool web app are:
- Average Daily Carbs is reported, but this will not include these zero bolus carbs
- Carb factors would appear to be correct when they are actually too low because of the zero bolus carbs from post meal carb treatments have been excluded
- Adding up the carbs displayed or exporting the data to calculate carbs will not match the PDM or reflect the total carbs consumed
- Carbs with a zero bolus during fitness will be missed so it will not be possible to refer back to carbs during previous fitness sessions
- This behavior is unexpected.  Carbs entered that result in a non-zero bolus are imported and displayed giving the impression that Tidepool is displaying the carbs logged in the PDM. Carbs are a critical part of treatment data and the expectation is that the data would capture all carb and insulin treatment logged.

These zero bolus carbs are shown when importing to Glooko and when viewing records in the PDM.

Perhaps a fix could be to change the wizard section of the filter (insuletSimulator.js:355) from:
```
if (bolus.normal === 0 && !bolus.expectedNormal) {
  return false;
}
else {
```  
to something like:
```
if (bolus.normal === 0 && !bolus.expectedNormal && event.carbInput === 0) {   
  return false;  
}
else {
```  

I'm not sure which zero bolus wizard events the filter is meant to exclude. I imagine it could be BG based wizard entries that result in a zero bolus because of IOB and thus have no clinical relevance and should be logged instead as smbg events. I don't believe that's the issue when carbs are entered with a zero bolus.

Steps to reproduce:
1. Navigate to the Omnipod PDM bolus screen: Home > Bolus
2. Can enter the BG reading or not and click yes or no
3. 'Are you going to eat?' click Yes
4. Set a non-zero number of carbs and confirm
5. Reduce the recommended bolus to 0.00 click enter and confirm and the PDM should return to the Home Screen

Viewing 'My records' in the PDM shows these carb events however importing the data from the PDM with the Tidepool uploader filters out these carb events and they do not appear in Tidepool.

**Comment by gniezen** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [[bug] Carbs not visualized for calculator bolus that includes carbs but no insulin delivery](https://trello.com/c/dZsluC3F/155-bug-carbs-not-visualized-for-calculator-bolus-that-includes-carbs-but-no-insulin-delivery)

**Comment by gniezen** [comment]
Hi Alex, thanks for the detailed issue report! :smile: We currently do not store any bolus wizard events where the delivered bolus is 0U, as we considered them to add too much noise to the data. We documented our reasoning in the [developer docs](http://developer.tidepool.org/data-model/device-data/types/bolus/normal.html#normal):

> To avoid noise in the data (especially as some insulin pumps make it very easy to deliver "doses" of 0 units when no additional dose is recommended given the PWD's current blood glucose and insulin on board), we have chosen in most cases not to allow the upload of boluses with a total delivered dose of 0 units; the value must be greater than 0.

That said, your use case to use the bolus wizard to track carbs makes sense, and you're right that the "Daily Average Carbs" widget suggests that the bolus wizard is used to track carbs, so we do want to take your suggestions on board. And as you pointed out, we can still filter out "noisy" boluses when there is no carb input.

**Comment by gniezen** [comment]
This was fixed in Uploader v2.13.0 realeased on 17 April 2019.
